### DISCUSSION TITLE
Fprime - available RTOS options

### DISCUSSION THREAD

**Discussion by arturodrt** [nasa/fprime#1947] [Q&A] [answered]
Hello everyone,

I am currently researching the most suitable RTOS for fprime. So far, I have discovered that it is possible to use fprime with freeRTOS.

Do you know if it is possible to use the PREEMPT_RT patch available for linux? 

In general, my question is which RTOS do you think are the most compatible or easiest to integrate with fprime?

**Comment by timcanham** [comment, marked as answer]
So far, we have integrated F Prime with Linux, Linux + RT patch, and VxWorks. There are others working on FreeRTOS and RTEMS. The easiest is probably Linux + RT, since it is easy to use with the existing Linux port.

**Comment by manas-narayan** [comment]
Hey @timcanham,
I am currently trying to deploy fprime with freeRTOS on my stm32f769 board. I found the laboratory's repo (fprime-freertos) on fprime-community but it prepares a executable for Linux. I want to know how can i build the same for my STM board. I am still struggling. Should i add the BSP package for board and create a toolchain file for my board. What approach should i take to implement this.

Also previously i was able to do the same with Zephyr RTOS and deployed it on my board.
